generator/internal/client_cc_generator.h

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019 Google Inc.  All rights reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef GOOGLE_API_CODEGEN_INTERNAL_CLIENT_CC_GENERATOR_H_
+#define GOOGLE_API_CODEGEN_INTERNAL_CLIENT_CC_GENERATOR_H_
+
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "src/google/protobuf/compiler/code_generator.h"
+#include "src/google/protobuf/io/zero_copy_stream.h"
+#include "src/google/protobuf/descriptor.h"
+#include "src/google/protobuf/descriptor.pb.h"
+#include "absl/strings/string_view.h"
+#include "gapic_generator.h"
+#include "printer.h"
+
+namespace pb = google::protobuf;
+
```

Comment:
It was a deliberate choice, as I was thinking about them as additional "pieces" of gapic_generator.cc (i.e. they should only ever be included from there), so didn't see value in adding cc files. I wouldn't be opposed to creating cc files though - and perhaps that is just a better practice to follow.

Reply:
It's one of the annoying things we have to think about when writing C/C++ :P
The main benefit I can think of in this case, where the code in question is very clearly internal-only and used in a single place, is to parallelize the build and make the linker do the heavy work instead of the compiler. That's a really premature optimization, though.

Reply:
I think I'll do it here, let's just follow standard convention unless there really is a good reason not to, and "saving lines of code by not having a .cc file" is probably not a good enough reason.